pkg/apis/garden/validation/validation_shoot.go

```diff
@@ -1776,10 +754,19 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *garden.ShootSpec, deletionTimesta
 	} else if newSpec.Cloud.AWS != nil {
 		allErrs = append(allErrs, validateK8SNetworksImmutability(oldSpec.Cloud.AWS.Networks.K8SNetworks, newSpec.Cloud.AWS.Networks.K8SNetworks, awsPath.Child("networks"))...)
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.AWS.Networks.VPC, oldSpec.Cloud.AWS.Networks.VPC, awsPath.Child("networks", "vpc"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.AWS.Networks.Internal, oldSpec.Cloud.AWS.Networks.Internal, awsPath.Child("networks", "internal"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.AWS.Networks.Public, oldSpec.Cloud.AWS.Networks.Public, awsPath.Child("networks", "public"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.AWS.Networks.Workers, oldSpec.Cloud.AWS.Networks.Workers, awsPath.Child("networks", "workers"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.AWS.Zones, oldSpec.Cloud.AWS.Zones, awsPath.Child("zones"))...)
+
+		if shouldValidateImmutability(newSpec.Cloud.AWS.Networks.Internal, oldSpec.Cloud.AWS.Networks.Internal) {
```

Comment:
I think we have to enforce that the order and the values stay the same. It must only be allowed to add new zones with new values, but not to swap them around.

Example:

Old Shoot has
```yaml
spec:
  cloud:
    aws:
      networks:
        internal:
        - 1.2.3.4/5
        - 5.6.7.8/9
```

It shall not be allowed to update to
```yaml
spec:
  cloud:
    aws:
      networks:
        internal:
        - 5.6.7.8/9
        - 1.2.3.4/5
```

However, it shall be allowed to

```yaml
spec:
  cloud:
    aws:
      networks:
        internal:
        - 1.2.3.4/5
        - 5.6.7.8/9
        - 10.11.12.13/14
```

(if the CIDRs don't overlap, of course, but this is already validated)

Reply:
The same applies for all other fields, especially the zones. It shall not be able to reorder them, just to add new zones.

Reply:
This is already considered by the enforce function, if the order is changed, validation will be required and thus Deepcopy will fail. Maybe add an additional test for that?

Reply:
Try out the following:

```golang
package main

import (
	"fmt"

	"k8s.io/apimachinery/pkg/util/sets"
)

func main() {
	old := []string{
		"1.2.3.4/5",
		"6.7.8.9/10",
	}
	new := []string{
		old[1],
		old[0],
		"11.12.13.14/15",
	}
	fmt.Println(shouldValidateImmutability(new, old)) // prints 'false' although it should print 'true'
}

func shouldValidateImmutability(new, old []string) bool {
	oldSet, newset := sets.NewString(old...), sets.NewString(new...)
	sizeDelta := len(new) - len(old)
	return sizeDelta < 0 || sizeDelta == 0 || !newset.IsSuperset(oldSet)
}
```

Reply:
okay, so as it is now, if the order is changed validation will be required, if additional zone is added and older values are changed (not re-ordered) is also considered, but what you mean is older values are reordered and additional zone is added, superset does not cover that unfortunately, now I see what you mean.

Reply:
I refined above example to make it more clear, sorry. 👍